storage/storage_dest.go

```diff
@@ -555,20 +593,22 @@ func (s *storageImageDestination) queueOrCommit(index int, info addedLayerInfo)
 	return nil
 }
 
-// getDiffIDOrTOCDigest returns the diffID for the specified digest or the digest for the TOC, if known.
-func (s *storageImageDestination) getDiffIDOrTOCDigest(uncompressedDigest digest.Digest) (digest.Digest, bool) {
+// getLayerID returns the diffID for the specified digest or the digest for the TOC, if known.
+func (s *storageImageDestination) getLayerID(uncompressedDigest digest.Digest, tocDigest digest.Digest) (digest.Digest, bool) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if d, found := s.diffOutputs[uncompressedDigest]; found {
-		return d.TOCDigest, found
+	if tocDigest != "" {
+		if d, found := s.tocDigests[uncompressedDigest]; found {
+			return d + "-toc", found
```

Comment:
This function claims to return a `digest.Digest`, a string like this isn’t a valid `digest.Digest`.

I think it’s better to make this return a string — and to make the name _clearly_ outside of the namespace of valid `digest.Digest` values. `-toc` is actually acepted in `digest.DigestRegexp` (it would _typically_ be rejected because the SHA-256 algorithm requires the digest value to be hexadecimal, but this doesn’t reject non-`sha256:` values… and being forward-compatible is cheap)

Move the `layerID.Hex()` call inside this function, to the non-TOC case.

Reply:
This was fixed here.